server/src/main/java/org/elasticsearch/index/mapper/GeoPointScriptFieldType.java

```diff
@@ -127,11 +135,75 @@ public Query distanceFeatureQuery(Object origin, String pivot, SearchExecutionCo
         double pivotDouble = DistanceUnit.DEFAULT.parse(pivot, DistanceUnit.DEFAULT);
         return new GeoPointScriptFieldDistanceFeatureQuery(
             script,
-            leafFactory(context)::newInstance,
+            valuesEncodedAsLong(context.lookup(), name(), leafFactory(context)::newInstance),
```

Comment:
I flipped all of the other queries to taking two doubles. I probably should flip this one as well, but it's a bit more complex so I'm waiting on the first review.

Reply:
Yes, this seems more complicated and I think this solution might make more sense. 

We need to quantize the values anyway so it behaves as in a normal field so what you are doing here makes lot of sense as we are only encoding values once.

Reply:
It would be ok to flip it too but we need to make sure we quantize the values to match the doc values of an index.